packages/remix-google-cloud-functions/server.ts

```diff
@@ -0,0 +1,133 @@
+import type {
+  Request as GcfRequest,
+  Response as GcfResponse,
+} from "@google-cloud/functions-framework";
+import type { 
+  AppLoadContext,
+  ServerBuild,
+  RequestInit as NodeRequestInit,
+ } from "@remix-run/node";
+import {
+  AbortController,
+  createRequestHandler as createRemixRequestHandler,
+  Headers as NodeHeaders,
+  Request as NodeRequest,
+  Response as NodeResponse,
+  writeReadableStreamToWritable
+} from "@remix-run/node";
+
+/**
+ * A function that returns the value to use as `context` in route `loader` and
+ * `action` functions.
+ *
+ * You can think of this as an escape hatch that allows you to pass
+ * environment/platform-specific values through to your loader/action.
+ */
+export interface GetLoadContextFunction {
+  (req: GcfRequest, res: GcfResponse): AppLoadContext;
+}
+
+export type RequestHandler = (
+  req: GcfRequest,
+  res: GcfResponse,
+) => Promise<void>;
+
+/**
+ * Returns a request handler for Express that serves the response using Remix.
+ */
+export function createRequestHandler({
+  build,
+  getLoadContext,
+  mode = process.env.NODE_ENV,
+}: {
+  build: ServerBuild;
+  getLoadContext?: GetLoadContextFunction;
+  mode?: string;
+}): RequestHandler {
+  let handleRequest = createRemixRequestHandler(build, mode);
+
+  return async (req: GcfRequest, res: GcfResponse) => {
+    try {
+      let request = createRemixRequest(req);
+      let loadContext =
+        typeof getLoadContext === "function"
+          ? getLoadContext(req, res)
+          : undefined;
+
+      let response = (await handleRequest(
+        request,
+        loadContext
+      )) as NodeResponse;
+
+      await sendRemixResponse(res, response);
+    } catch (error) {
+        console.error(error);
+        await sendRemixResponse(res ,new NodeResponse("Internal Error", { status: 500 }));
+    }
```

Comment:
Other adapters don't have a `try`/`catch` block here, so let's not do it either.

```suggestion
    let request = createRemixRequest(req);
    let loadContext =
      typeof getLoadContext === "function"
        ? getLoadContext(req, res)
        : undefined;

    let response = (await handleRequest(
      request,
      loadContext
    )) as NodeResponse;

    return sendRemixResponse(res, response);
```

Reply:
also the express adapter has `next(error)` which serves a similar purpose but as previously discussed on this PR google cloud functions doesn't have an equivalent

Reply:
`@remix-run/deno` is a runtime, not an adapter.

`@remix-run/express` indeed has a `try`/`catch` in `createRequestHandler`, but that's because `express` doesn't support async functions, so we have to pass along the error manually using `next()`.

Reply:
> Without this errors go unrecorded and the client http request will just hang

although this is a fair point, I'm not sure  what the best approach would be in that case, @MichaelDeBoey ?

Reply:
in the adapter I was writing, I was originally using the firebase-functions library and throwing a specific error type, but this didn't seem to exist in the google cloud functions library, sec I'll go dig it out

Reply:
Here's the `firebase-functions` documentation on throwing `functions.https.HttpsError`, though as above I couldn't find an equivalent for `@google-cloud/functions-framework`


https://firebase.google.com/docs/functions/callable#handle_errors

Reply:
I think we need a unit test here that shows the hanging or handling of the error depending on approach